src/libPMacc/include/types.h

```diff
@@ -60,6 +61,16 @@ typedef long long int int64_cu;
 #define DINLINE __device__ __forceinline__
 #define HINLINE __host__ inline
 
+/**
+ * CUDA architecture version (aka PTX ISA level)
+ * 0 for host compilation
+ */
+#ifndef __CUDA_ARCH__
```

Comment:
that introduces a general new naming.
I would prefer using it consistent over the whole code or alternatively not introducing it (`defined && >` works, too).